operator/pkg/controller/istiocontrolplane/istiocontrolplane_controller.go

```diff
@@ -324,6 +329,20 @@ func (r *ReconcileIstioOperator) Reconcile(request reconcile.Request) (reconcile
 		return reconcile.Result{}, err
 	}
 
+	shouldVerifyInstall, _ := os.LookupEnv("VERIFY_CR_INSTALL")
+	if shouldVerifyInstall == "true" {
+		scope.Infof("Verifying installation %s/%s (revision=%s)",
+			iopMerged.Namespace, iopMerged.Name, iopMerged.Spec.Revision)
+
+		// TODO(su225): How do I get kubeconfig and context?
+		// It is currently broken. Do something about it.
```

Comment:
this is because you cannot use kubeconfig/context in operator. operator needs to use its own role and creds in the dynamic kube client. unfortunately this means that verifier may need to be significantly refactored to be able to support a kube client either derived from kubeconfig/context (for cli case) or from the dynamically generated kube client (in operator case).

Reply:
In the end, all we need is a client to API server. It would be better for the caller to supply kube-client instead of letting verifier to figure it out. That should have been the case. Verifier should not worry about kubernetes related stuffs. Yes, I agree that it has to be refactored - especially the parts with Visitor.

Reply:
As discussed in chat, I removed verification code from operator parts.